pkg/envoy/agent_test.go

```diff
@@ -127,10 +164,46 @@ func TestRecovery(t *testing.T) {
 		<-ctx.Done()
 		return nil
 	}
-	a := NewAgent(TestProxy{run: start}, 0)
+	a := NewAgent(TestProxy{run: start}, 0, 0, "", 0, 0, 0)
 	go func() { a.Run(ctx) }()
 
 	// make sure we don't try to reconcile twice
 	<-time.After(100 * time.Millisecond)
 	cancel()
 }
+
+func TestActiveConnections(t *testing.T) {
+	cases := []struct {
+		name     string
+		stats    string
+		expected int
+	}{
+		{
+			"invalid stats",
+			invalidStats,
+			-1,
```

Comment:
If someone disables all stats we should have sane behavior. Probably the old behavior? wdyt?

Reply:
Either that, or we add this to the stats that are always and automatically added.
I think having these stats is important anyway.
Several users enable these in production.

Reply:
I have added them by default. Some one can not disable them I guess.